shared/src/components/buttons/async-button.cjsx

```diff
@@ -40,7 +40,7 @@ module.exports = React.createClass
 
   checkForTimeout: ->
     {isWaiting} = @props
-    @setState(isTimedout: true, delayTimeout: null) if isWaiting and @isMounted()
+    @setState(isTimedout: true, delayTimeout: null) if isWaiting
```

Comment:
Could removing this `@isMounted()` check cause exceptions when a network request begins and the student closes the window before the server responds?